[dynamo] updated additional info and changed json to yml

Sidharth123-cpu opened the pull request:
This PR switches from the reigstry being in JSON format to YAML format in order to store code samples for additional info. I also added a new feature where anytime --additional info is ran, the terminal opens up a nano and the dev can put any content they want, which will then be stored under additional info for that GB_type.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #158327



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

Commit:
[dynamo] updated additional info and changed json to yaml

[ghstack-poisoned]

Commit:
Update on "[dynamo] updated additional info and changed json to yaml"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Commit:
Update on "[dynamo] updated additional info and changed json to yml"

This PR switches from the reigstry being in JSON format to YAML format in order to store code samples for additional info. I also added a new feature where anytime --additional info is ran, the terminal opens up a nano and the dev can put any content they want, which will then be stored under additional info for that GB_type.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Commit:
Update on "[dynamo] updated additional info and changed json to yml"

This PR switches from the reigstry being in JSON format to YAML format in order to store code samples for additional info. I also added a new feature where anytime --additional info is ran, the terminal opens up a nano and the dev can put any content they want, which will then be stored under additional info for that GB_type.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Commit:
Update on "[dynamo] updated additional info and changed json to yml"

This PR switches from the reigstry being in JSON format to YAML format in order to store code samples for additional info. I also added a new feature where anytime --additional info is ran, the terminal opens up a nano and the dev can put any content they want, which will then be stored under additional info for that GB_type.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

williamwen42 commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -2,11 +2,22 @@
+import tempfile
 from pathlib import Path
 
+from ruamel.yaml import YAML
```
Why are we using `raumel`? PyTorch already has a dependency on `pyyaml` - why can't we use that?

williamwen42 commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -15,20 +26,71 @@ def get_source_segment(source, node):
     return f"GB{(max(ids, default=0) + 1):04d}"
 
 
+def get_editor_input(prompt_text="", existing_content=""):
```
do you ever call this function with `existing_content` set?

Commit:
Update on "[dynamo] updated additional info and changed json to yml"

This PR switches from the reigstry being in JSON format to YAML format in order to store code samples for additional info. I also added a new feature where anytime --additional info is ran, the terminal opens up a nano and the dev can put any content they want, which will then be stored under additional info for that GB_type.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames

[ghstack-poisoned]

Sidharth123-cpu commented on tools/dynamo/gb_id_mapping.py:
```diff
@@ -269,29 +349,48 @@ def cmd_update_gb_type(
+            f"Opening editor for additional information for '{matching_call['gb_type']}'..."
+        )
+
+        editor_content = get_editor_input(
```
So initially, I thought about this, hence, why originally the plan was to have existing_content, however, I figured the nano would get too clustered with existing additional_info if there are large chunks of code, so everytime they open up the nano, it'll be something new they can put just to make things cleaner. If they want to edit existing additional_info they can manually do it in the yml file, or put it in the manual section of the markdown file.

williamwen42 reviewed (commented):
`raumel` may need to be added to a dependency file somewhere now - please double check with devinfra to see if this is the case.

Note: As discussed earlier, I found way to avoid using `raumel`
